scenario/state.py

```diff
@@ -140,6 +141,79 @@ def copy(self) -> "Self":
         return copy.deepcopy(self)
 
 
+@dataclasses.dataclass(frozen=True)
+class CloudCredential:
+    auth_type: str
+    """Authentication type."""
+
+    attributes: Dict[str, str] = dataclasses.field(default_factory=dict)
+    """A dictionary containing cloud credentials.
+
+    For example, for AWS, it contains `access-key` and `secret-key`;
+    for Azure, `application-id`, `application-password` and `subscription-id`
+    can be found here.
+    """
+
+    redacted: List[str] = dataclasses.field(default_factory=list)
+    """A list of redacted secrets."""
+
+    def to_ops_cloud_credential(self):
```

Comment:
Just a couple of things about these `to_ops` methods:

* We should make this private (i.e., start with `_`), as it's an implementation detail of Scenario.
* We can probably drop the `_cloud_credential` part of the name. Do other dataclasses in Scenario do a similar thing, and is there a naming convention for these?
* Can we do something like `ops.CloudCredential(**dataclasses.asdict(self))`? Maybe similar with `CloudSpec`, though maybe that's trickier given the nested `credential` field.

Reply:
The only "to_ops" I know of is in the (unmerged) Notices PR. We did have a bit of a [naming discussion there](https://github.com/canonical/ops-scenario/pull/108#discussion_r1550664658).

`BindAddress` and `Network` have a `hook_tool_output_fmt` method. Probably this (and also the Notices PR) should follow that, although I agree that it's better if this is private (we have changing the existing ones to private proposed for 7.0). I'm guessing these are the ones that Pietro was referring to earlier.

There's also the global `_status_to_entitystatus` (although that's converting the other way around).

Reply:
I have made them private because I think it makes sense: it wouldn't be used directly by the user. In the beginning I thought I had one in the unit test but I was wrong.

Reply:
Started a discussion in Matrix about `_to_ops`, `_to_ops_object` and `_to_ops_cloud_credential`.

Reply:
According to our discussion in Matrix, we will use `_to_ops` to try to keep it consistent. Resolving now.